src/utilities/storage.ts

```diff
@@ -0,0 +1,16 @@
+import { HabitSlice } from "../app/features/habit/habitTypes";
+
+export class Storage {
+  habit: HabitSlice | null;
+  constructor() {
+    const savedHabit = localStorage.getItem("habit");
+
+    this.habit = savedHabit ? JSON.parse(savedHabit) : null;
+  }
```

Comment:
why you need to store `habit` here? It's better to have such classes stateless. Currently the following will produce an incorrect result while properly using the API:
```ts
const habit = storage.habit;
storage.updateStorage()
const updatedHabit = storage.habit
```